packages/embark/src/lib/modules/code_generator/index.js

```diff
@@ -332,9 +337,11 @@ class CodeGenerator {
             this.logger.error(__('Error creating a symlink to EmbarkJS'));
             return next(err);
           }
-          embarkjsCode += `\nconst EmbarkJS = require("${symlinkDest}").default;`;
-          embarkjsCode += "\nexport default EmbarkJS;";
-          embarkjsCode += "\nglobal.EmbarkJS = EmbarkJS";
+          embarkjsCode += `\nconst EmbarkJS = require("${symlinkDest}").default || require("${symlinkDest}");`;
+          embarkjsCode += "\nglobal.EmbarkJS = EmbarkJS;";
+          embarkjsCode += `\nif (typeof window === 'undefined') {
+            global.window = {}; // Make sure window is not undefined in the case of a node use
```

Comment:
What if a user writing code targeting both node and browser is using global `window` detection, like you did on the line above, to differentiate environments?  Creating the global could cause their code to behave unexpectedly, or that's a concern I have.

Reply:
Ah yes, it's a good concern. To be honest, I was not really sure how to handle that. The problem this fixed is that `EmbarkJS.Blockchain` checks `window.ethereum` and it made it crash.
I guess I'd be better off fixing at that spot and do the check on `window` there.
Hopefully we don't use `window anywhere else